backend/corpora/dataset_processing/remaster_cxg.py

```diff
@@ -0,0 +1,250 @@
+#!/usr/bin/env python3
+import logging
+import subprocess
+
+from backend.corpora.dataset_processing.process import (
+    get_bucket_prefix,
+)
+
+import time
+
+import numpy as np
+import psutil
+import tiledb
+
+
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+def process(dataset_id: str, cellxgene_bucket: str):
+    """
+    1. Download the labeled dataset from the artifact bucket
+    2. Convert the labeled dataset to CXG
+    3. Upload the CXG to the cellxgene bucket
+    :param dataset_id:
+    :param artifact_bucket:
+    :param labeled_h5ad_filename:
+    :param cellxgene_bucket:
+    :return:
+    """
+
+    # Careful with old datasets - the ones that do not have a UUID
+    # cxg_filename = f"{dataset_id}.cxg"
+
+    # Download the cxg from the bucket
+    bucket_prefix = get_bucket_prefix(dataset_id)
+    object_key = f"{bucket_prefix}.cxg"
+    path = f"s3://{cellxgene_bucket}/{object_key}/X"
+    local_path = "/cxg"
+    logger.info(f"Downloading {path} to {local_path}/X_old")
+
+    download_command = ["aws", "s3", "sync", path, f"{local_path}/X_old"]
+    # Let errors fail the pipeline
+    subprocess.run(download_command, check=True)
+
+    kind = "auto"
+    obs_extent = 256
+    var_extent = 2048
+    cell_order = "row"
+    tile_order = "col"
+    capacity = 128000
+    compression = 22
+    target_array = "X_new"
+    source_array = "X_old"
+    sparse_threshold = 25.0
+
+    compute(
+        local_path,
+        obs_extent,
+        var_extent,
+        source_array,
+        target_array,
+        kind,
+        sparse_threshold,
+        compression,
+        cell_order,
+        tile_order,
+        capacity,
+    )
+
+    upload_command = ["aws", "s3", "sync", f"{local_path}/X_new", path]
+    subprocess.run(upload_command, check=True)
+
+def compute(
+    cxg,
+    obs_extent,
+    var_extent,
+    source_array,
+    target_array,
+    kind,
+    sparse_threshold,
+    compression,
+    cell_order,
+    tile_order,
+    capacity,
+):
+    X_extent = [obs_extent, var_extent]
+
+    ctx = create_fast_ctx(
+        {
+            "py.init_buffer_bytes": 32 * 1024**3,
+            "sm.tile_cache_size": 2 * 1024**3,
```

Comment:
is overriding with a fixed constant preferable to the default (which is 50% of the environments memory)?  See `create_fast_ctx` for the default.

Reply:
Honestly, I am not sure what's the rationale behind these values, I just got them from Ben's script. Since the default isn't used anywhere else, we should probably remove it and keep this value.